org.mwc.debrief.lite/src/test/java/org/mwc/debrief/lite/utils/TestUtils.java

```diff
@@ -15,10 +15,85 @@
 package org.mwc.debrief.lite.utils;
 
 /**
+ * 
  * @author Ayesha
  *
  */
-public class TestUtils
-{
+import java.awt.*;
+import javax.swing.*;
 
+public class TestUtils {
+
+  static int counter;
+
+  public static Component getChildNamed(Component parent, String name) {
+
+    // Debug line
+    //System.out.println("Class: " + parent.getClass() +
+    //    " Name: " + parent.getName());
+
+    if (name.equals(parent.getName())) { return parent; }
+
+    if (parent instanceof Container) {
+      Component[] children = (parent instanceof JMenu) ?
+          ((JMenu)parent).getMenuComponents() :
+          ((Container)parent).getComponents();
+
+      for (int i = 0; i < children.length; ++i) {
+        Component child = getChildNamed(children[i], name);
+        if (child != null) { return child; }
+      }
+    }
+    
+    return null;
+  }
+
+  public static Component getChildIndexed(
+      Component parent, String klass, int index) {
+    counter = 0;
+
+    // Step in only owned windows and ignore its components in JFrame
+    if (parent instanceof Window) {
+      Component[] children = ((Window)parent).getOwnedWindows();
+
+      for (int i = 0; i < children.length; ++i) {
+        // take only active windows
+        if (children[i] instanceof Window &&
+            !((Window)children[i]).isActive()) { continue; }
+
+        Component child = getChildIndexedInternal(
+            children[i], klass, index);
+        if (child != null) { return child; }
+      }
+    }
+
+    return null;
+  }
+
+  private static Component getChildIndexedInternal(
+      Component parent, String klass, int index) {
+
+    // Debug line
+    //System.out.println("Class: " + parent.getClass() +
+    //    " Name: " + parent.getName());
+
+    if (parent.getClass().toString().endsWith(klass)) {
+      if (counter == index) { return parent; }
+      ++counter;
+    }

```

Comment:
@ayeshare - I'm nervous of the use of this static field. It doesn't look thread safe.  Could you introduce some comments to explain what's happening in this method?

Hmm, a more robust strategy could be to pass `AtomicInteger` down through the call chain.  The value of the element can be updated.